pkg/fakerp/clusterstatus.go

```diff
@@ -0,0 +1,59 @@
+package fakerp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/plugin"
+	adminclient "github.com/openshift/openshift-azure/pkg/util/azureclient/openshiftmanagedcluster/admin"
+)
+
+// handleClusterStatus handles admin requests for the status of control plane pods
+func (s *Server) handleClusterStatus(w http.ResponseWriter, req *http.Request) {
+	cs := s.read()
+	if cs == nil {
+		s.internalError(w, "Failed to read the internal config")
+		return
+	}
+
+	config, err := GetPluginConfig()
+	if err != nil {
+		s.internalError(w, fmt.Sprintf("Failed to configure plugin: %v", err))
+		return
+	}
+	p, errs := plugin.NewPlugin(s.log, config)
+	if len(errs) > 0 {
+		s.internalError(w, fmt.Sprintf("Failed to configure plugin: %v", err))
+		return
+	}
+
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, api.ContextKeyClientID, cs.Properties.ServicePrincipalProfile.ClientID)
+	ctx = context.WithValue(ctx, api.ContextKeyClientSecret, cs.Properties.ServicePrincipalProfile.Secret)
+	ctx = context.WithValue(ctx, api.ContextKeyTenantID, cs.Properties.AzProfile.TenantID)
+
+	// TODO: Perhaps get the namespace list from the request's query parameter
+	namespaces := []string{"kube-system", "openshift-node"}
+
+	pods, err := p.ClusterStatus(ctx, cs, namespaces)
+	if err != nil {
+		s.internalError(w, fmt.Sprintf("Failed to fetch cluster status: %v", err))
+		return
+	}
+
+	response := adminclient.OpenShiftManagedClustersStatus{
```

Comment:
don't introduce a new type.  write the []byte returned from the plugin with a content-type of application/json